source/vibe/http/common.d

```diff
@@ -469,20 +488,25 @@ final class ChunkedOutputStream : OutputStream {
 	{
 		assert(!m_finalized);
 		if( m_buffer.data.length > 0 ) flush();
+		ubyte[] buffer = null;
+		auto getBytes = (size_t size) {
+			if (buffer is null || buffer.length < size)
+				buffer = new ubyte[](size);
+				return buffer[0..size];
+			};
```

Comment:
Actually, I'm not sure if that would still be reasonable. I thought about writing a method that requests N bytes in the buffer and returns the according range in the buffer (so one could `stream.read(appendBytes(N))`), but the `Appender` struct seems not to be made for this case.
To be able to conveniently append data to the buffer, one seems to basically always have a range of items already allocated.

This could work, but I'm not sure if it is still efficient:
```{D}
private ubyte[] appendBytes(ulong maxSize)
{
    if (m_maxBufferSize > 0 && m_maxBufferSize < m_buffer.data.length + maxSize)
        maxSize = m_maxBufferSize - min(m_buffer.data.length, m_maxBufferSize);
    auto pos = m_buffer.data.length;
    for (ulong i = 0; i < maxSize; ++i)
        m_buffer.put(0);
    return m_buffer.data[pos..pos+maxSize];
}
```

Then I could basically do:
```{D}
auto bytes = appendBytes(sz);
if (bytes.length > 0)
    data.read(bytes);
else
    flush();
```

And for the `write(in ubyte[])` method it would be:
```{D}
while (bytes.length > 0) {
    auto sz = bytes.length;
    auto dst = appendBytes(sz);
    if (dst.length > 0)
    {
        dst[0..sz] = bytes[0..sz];
        bytes = bytes[sz..$];
    }
    else
        flush();
}
```

Do you perhaps have a better idea? Another variant could be to use initialize the appender with the according block size and use it like a normal array as long as `m_maxBufferSize > 0` (basically only using `put()` for unbounded chunk sizes).